### PR TITLE
helm: add more addons for gateway-addons-helm

### LINKED ISSUE #3355: Add-on / Extended Helm Chart
*Description*:
>Describe the issue.

Consider introducing a add-on / helm chart that may be common while Installing Envoy Gateway
e.g. Monitoring components highlighted in 
https://gateway.envoyproxy.io/v1.0.1/tasks/observability/proxy-observability/ & 
https://gateway.envoyproxy.io/v1.0.1/tasks/observability/grafana-integration/

Relates to https://github.com/envoyproxy/gateway/pull/2260

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

xref #3355

### CODE REVIEW

**Comment by shawnh2** [comment]
Once the `gateway-addons-helm` has been added to OCI Registry successfully, we can update the [observability doc](https://gateway.envoyproxy.io/latest/tasks/observability/) and makefile cmds.

File: charts/gateway-addons-helm/Chart.yaml [hunk 1]
```diff
@@ -26,8 +26,20 @@ sources:
...
 dependencies:
   - name: prometheus
-    version: 23.1.0
+    version: 25.21.0
     repository: https://prometheus-community.github.io/helm-charts
   - name: grafana
     repository: https://grafana.github.io/helm-charts
-    version: 6.58.4
+    version: 8.0.0
+  - name: fluent-bit
```
Comment (zirain) [inline charts/gateway-addons-helm/Chart.yaml:34]: IMO, should add `condition` which give user ability to disable subchart.

Comment (shawnh2) [reply charts/gateway-addons-helm/Chart.yaml:34]: good point!

**Review by zirain — APPROVED** [review]

File: tools/make/docs.mk [hunk 2]
```diff
@@ -111,10 +111,10 @@ docs-release-gen:
 docs-check-links:
 	@$(LOG_TARGET)
 	# Check for broken links, right now we are focusing on the v1.0.0
-	# github.com does not allow access too often, there're a lot of 429 errors
+	# github.com does not allow access too often, there are a lot of 429 errors
 	# TODO: find a way to remove github.com from ignore list
 	# TODO: example.com is not a valid domain, we should remove it from ignore list
-	linkinator site/public/ -r --concurrency 25 -s "github.com example.com _print v0.6.0 v0.5.0 v0.4.0 v0.3.0 v0.2.0"
+	linkinator site/public/ -r --concurrency 25 --skip "github.com example.com github.io _print v0.6.0 v0.5.0 v0.4.0 v0.3.0 v0.2.0"
```
Comment (arkodg) [inline tools/make/docs.mk:117]: intentionally part of this PR ?

File: charts/gateway-addons-helm/values.yaml [hunk 3]
```diff
@@ -53,3 +61,173 @@ prometheus:
...
+    prometheus.io/scrape: "true"
+    prometheus.io/port: "2020"
+    prometheus.io/path: /api/v1/metrics/prometheus
+    fluentbit.io/exclude: "true"
+  ## https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/configuration-file
+  config:
+    service: |
+      [SERVICE]
+          Daemon Off
+          Flush {{ .Values.flush }}
```
Comment (arkodg) [inline charts/gateway-addons-helm/values.yaml:82]: these templates are being filled by `fluent-bit` or our tooling ?

File: charts/gateway-addons-helm/values.yaml [hunk 4]
```diff
@@ -1,5 +1,12 @@
+tags:
```
Comment (arkodg) [inline charts/gateway-addons-helm/values.yaml:1]: curious why we need this ?

File: tools/make/docs.mk (hunk 2 above)
Comment (shawnh2) [reply tools/make/docs.mk:117]: yes, the repository of open-telemetry collector: `https://open-telemetry.github.io/opentelemetry-helm-charts` is 404, but works for helm

File: charts/gateway-addons-helm/values.yaml (hunk 3 above)
Comment (shawnh2) [reply charts/gateway-addons-helm/values.yaml:82]: these values are directly copied from https://github.com/envoyproxy/gateway/blob/main/examples/fluent-bit/helm-values.yaml, same for others.

the values under example dir will be removed within follow-up PR.

File: charts/gateway-addons-helm/values.yaml (hunk 4 above)
Comment (shawnh2) [reply charts/gateway-addons-helm/values.yaml:1]: to provide users more options to play with, also indicates which component is all about metrics or logging etc.

Comment (arkodg) [reply charts/gateway-addons-helm/values.yaml:1]: ah TIL about this https://helm.sh/docs/topics/charts/#tags-and-condition-fields-in-dependencies :)

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by cnvergence — APPROVED** [review]

**Pull request merged**